src/dhtproto/client/NotifierTypes.d

```diff
@@ -89,3 +134,65 @@ public struct RequestKeyInfo
             this.request_id, this.key);
     }
 }
+
+import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+/*******************************************************************************
+
+    Mixin for method to serialize a record value.
+
+    Params:
+        dst = pointer to the buffer to be serialized into
+
+    TODO: deprecated, replace with implementation in swarm
+
+*******************************************************************************/
+
+template SerializeMethod ( alias dst )
+{
+    import ocean.util.serialize.contiguous.Contiguous;
+    import ocean.util.serialize.Version;
+    import ocean.util.serialize.contiguous.Serializer;
+    import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+    /***************************************************************************
+
+        Serializes `src` into `dst`, using a version decorater, if required.
+
+        Params:
+            T = type of struct to serialize
+            src = instance to serialize
+
+    ***************************************************************************/
+
+    public void serialize ( T ) ( T src )
+    {
+        static if ( Version.Info!(T).exists )
+        {
+            if ( client_serializer_version_decorator is null )
+                client_serializer_version_decorator = new VersionDecorator;
+
+            client_serializer_version_decorator.store!(T)(src, *dst);
```

Comment:
I don't think you actually need to bother with allocating a singleton `VersionDecorator` instance: `VersionDecorator.store` will do, as it is a `static` method:
https://github.com/sociomantic-tsunami/ocean/blob/e9b693dc46d6d6b80bd54244b004b0ecd7679c64/src/ocean/util/serialize/contiguous/MultiVersionDecorator.d#L107

The underlying buffer of `VersionDecorator` is only used for deserialization IIRC.

Reply:
Ah that's good to know :+1: .

Reply:
Updated.